internal/server/event_server.go

```diff
@@ -69,4 +79,22 @@ func (s *EventServer) ListenAndServe(stopCh <-chan struct{}) {
 	} else {
 		s.logger.Info("Event server stopped")
 	}
+
+	return nil
+}
+
+func eventKeyFunc(r *http.Request) (string, error) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return "", err
+	}
+	defer r.Body.Close()
+
+	event := &recorder.Event{}
+	err = json.Unmarshal(body, event)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("event/%s/%s", event.InvolvedObject.String(), event.Severity), nil
```

Comment:
We need to do a checksum of the name, namespace, kind and event message. As it this, means that we drop health checks after apply. Please see my proposal in the linked issues.

Reply:
This does not solve the problem of a even going from one state to the other as the message will still be the same. One option is to just ignore it and accept it as a limitation.

This test fails but according to me should pass.
https://github.com/fluxcd/notification-controller/blob/7c13a5f58ca0a46fbdc8411346c48788700543ee/internal/server/event_server_test.go#L80